### PR TITLE
clue plugin: accept spirit angler outfit as clue requirement

### LINKED ISSUE #13420: Spirit Angler's doesn't list as counting for clue scrolls
**Describe the bug**
I have three pieces of Spirit Anglers (headband, torso, legs) and one piece of standard Angler's (boots). The clue helper plugin did not recognize this as meeting the requirements for Sherlock's "Fish an anglerfish in your finest gear" step, but the game did count it when I completed the clue scroll.

**To Reproduce**
- Get the Angler set Sherlock step
- Look at UI
- Ignore the plugin and do the step anyway

**Expected behavior**
I wanted the plugin to say that I have the required items when I have some or all pieces of Spirit Angler's in place of the normal Angler Outfit.

**Environment (please complete the following information):**
 - Windows
 - RuneLite version: 1.7.3.1
 - Launcher version: 2.0.2

### PR DESCRIPTION
**Pull request by molo-pl**
Fixes #13420
Unfortunately I wasn't able to test this change manually, since it may be quite difficult to obtain a specific master clue skill challenge, and I don't know if there are any dev tools which would allow faking/mocking it.

### CODE REVIEW

**Comment by Hydrox6** [comment]
Either use `::clue` to set the clue text to what you input, or just force the specific clue text in the function that chooses the clue from the text

**Comment by molo-pl** [comment]
Thanks, seems to be working:

screenshot

**Comment by Nightfirecat** [comment]
I think a test is actually overkill for this. We have used a similar setup for other outfit sets without issue, I recommend just changing the Angler's outfit entry to match that. (see the graceful set skill challenge clue)

**Comment by molo-pl** [comment]
I can remove the test, but is it really causing any harm? Usually it's integration / end-to-end tests tests which cause problems - they tend to fail intermittently, they are difficult to maintain, they prolong build pipeline and consume resources. None of that is true about unit tests. Also, in my opinion not having tests for a piece of code doesn't sound like a good precedent for future changes.

**Comment by Hydrox6** [comment]
Like Nightfirecat said, the test is overkill, and you're also making larger-than-needed changes to the code to facilitate that test (the `ANGLER_OUTFIT` static requirement is extraneous)

File: runelite-client/src/main/java/net/runelite/client/plugins/cluescrolls/clues/SkillChallengeClue.java [hunk 1]
```diff
@@ -188,7 +188,11 @@
...
 		new SkillChallengeClue("Dissect a sacred eel.", item(ItemID.KNIFE), any("Fishing rod", item(ItemID.FISHING_ROD), item(ItemID.PEARL_FISHING_ROD)), item(ItemID.FISHING_BAIT)),
 		new SkillChallengeClue("Kill a lizardman shaman."),
-		new SkillChallengeClue("Catch an Anglerfish.", "angle for an anglerfish whilst sporting the finest fishing gear.", true, any("Fishing rod", item(ItemID.FISHING_ROD), item(ItemID.PEARL_FISHING_ROD)), item(ItemID.SANDWORMS), all("Angler's outfit", item(ItemID.ANGLER_HAT), item(ItemID.ANGLER_TOP), item(ItemID.ANGLER_WADERS), item(ItemID.ANGLER_BOOTS))),
+		new SkillChallengeClue("Catch an Anglerfish.", "angle for an anglerfish whilst sporting the finest fishing gear.", true, any("Fishing rod", item(ItemID.FISHING_ROD), item(ItemID.PEARL_FISHING_ROD)), item(ItemID.SANDWORMS), all("Angler's outfit",
+			any("", item(ItemID.ANGLER_HAT), item(ItemID.SPIRIT_ANGLER_HEADBAND)),
+			any("", item(ItemID.ANGLER_TOP), item(ItemID.SPIRIT_ANGLER_TOP)),
+			any("", item(ItemID.ANGLER_WADERS), item(ItemID.SPIRIT_ANGLER_WADERS)),
+			any("", item(ItemID.ANGLER_BOOTS), item(ItemID.SPIRIT_ANGLER_BOOTS)))),
```
Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/plugins/cluescrolls/clues/SkillChallengeClue.java:195]: To match the format of other outfits allowing alternative pieces. (I'll commit this via GitHub's UI)
```suggestion
		new SkillChallengeClue("Catch an Anglerfish.", "angle for an anglerfish whilst sporting the finest fishing gear.", true, any("Fishing rod", item(ItemID.FISHING_ROD), item(ItemID.PEARL_FISHING_ROD)), item(ItemID.SANDWORMS),
			all("Angler's outfit",
				any("", item(ItemID.ANGLER_HAT), item(ItemID.SPIRIT_ANGLER_HEADBAND)),
				any("", item(ItemID.ANGLER_TOP), item(ItemID.SPIRIT_ANGLER_TOP)),
				any("", item(ItemID.ANGLER_WADERS), item(ItemID.SPIRIT_ANGLER_WADERS)),
				any("", item(ItemID.ANGLER_BOOTS), item(ItemID.SPIRIT_ANGLER_BOOTS)))),
```

**Pull request merged**